test/document_test.rb

```diff
@@ -334,6 +338,11 @@ def setup
     assert_equal foobaz.foo, {'foo' => 'bar', 'bar' => 'bar'}
   end
 
+  test 'default should populate before save' do
+    test_foo = TestFoo.new
+    assert_equal 'FOO', test_foo.doc['default_test_property']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.